core/src/main/java/org/ironjacamar/core/connectionmanager/pool/AbstractPool.java

```diff
@@ -397,4 +422,49 @@ else if (txCM.getTransactionIntegration().isConnectableResource(mc))
     * @return The instance
     */
    protected abstract ManagedConnectionPool createManagedConnectionPool(Credential credential);
+
+   /**
+    * Prefill the connection pool
+    *
+    * @param credential the pool credential
+    */
+   @Override
+   public void prefill()
+   {
+
+      if (poolConfiguration.isPrefill())
+      {
+         ManagedConnectionPool mcp = pools.get(getPrefillCredential());
```

Comment:
yup it has, but there is no way to understand if it is an already present mcp or created at the moement of the call. See comment below